bench/textx/json.py

```diff
@@ -0,0 +1,50 @@
+from textx import metamodel_from_file
+from os.path import join, dirname
+import collections
+from bench import helpers
+
+
+def compile():
+    class Member:
+        def __init__(self, key, value, parent=None):
+            self.pair = (key, value)
+
+        def __repr__(self):
+            return str(self.pair)
+
+        def __iter__(self):
+            return iter(self.pair)
+
+    class Object(collections.UserDict):
+        def __init__(self, members, parent=None):
+            self.data = {key: value for key, value in members}
+
+    class Array(collections.UserList):
+        def __init__(self, values, parent=None):
+            self.data = list(values)
```

Comment:
It's not actually necessary to use the UserDict/UserList classes. The following works, too:

```python
    class Object(dict):
        def __init__(self, members, parent=None):
            self.update((m.key, m.value) for m in members)

    class Array(list):
        def __init__(self, values, parent=None):
            self.extend(values)
```

And note that there's no need for importing `collections` and the `Members` class can be completely removed.

Reply:
To clarify: You can remove the `collections` import if you use the simplified classes. And there was no speed improvement to using these classes over the `User*` ones, but the 0.2s was gained by removing the custom `Member` class and just using the default textX one in the `Object.__init__()` method.